src/pages/Graph/GraphToolbar/GraphSettings.tsx

```diff
@@ -151,47 +226,84 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps, GraphSetting
       }
     ];
 
-    const { isOpen } = this.state;
-
-    const selection = visibilityLayers
-      .filter((item: VisibilityLayersType) => item.value)
-      .concat(badges.filter((item: VisibilityLayersType) => item.value))
-      .map((item: VisibilityLayersType) => item.labelText);
-
     return (
-      <Select
-        placeholderText="Display"
-        onToggle={this.onToggle}
-        onSelect={() => undefined}
-        isExpanded={isOpen}
-        variant={SelectVariant.checkbox}
-        isGrouped={true}
-        selections={selection}
+      <BoundingClientAwareComponent
+        className={containerStyle}
+        maxHeight={{ type: PropertyType.VIEWPORT_HEIGHT_MINUS_TOP, margin: marginBottom }}
```

Comment:
I'm glad this thing has been useful :)

Reply:
super useful! I still don't know how it works.